### ISSUE TITLE
When dragging sprites, mouse position is wrongly calculated in a CSS-scaled game

### ISSUE THREAD

**Issue by fariazz** [photonstorm/phaser#3666]
If you scale the game with CSS to show the entire game, create a sprite which is draggable, and make it follow the mouse when dragging, you will see that the position of the mouse doesn't match that of the sprite.

** Video: **   https://www.screencast.com/t/oeqtjb6SMxY

** Steps to repeat: **

- download this example:   http://labs.phaser.io/edit.html?src=src\input\dragging\enable%20for%20drag.js

- add this CSS code to the top of the document:

```
<style>
      html, body {
        height: 100%;
        background: black;
        margin: 0;

      }

      canvas {
        width: 100%;
        height: 100%;
        object-fit: contain;
      }
    </style>
```

**Comment by fariazz** [comment]
Update: this only happens when the height is scaled. If just the width is scaled, it works as expected (sprite follows the mouse position correctly).

**Comment by photonstorm** [comment]
Hmm, I can't replicate that using the examples on labs:

http://labs.phaser.io/css.html?src=src\input\dragging\enable%20for%20drag.js

Is this browser specific?

**Comment by photonstorm** [comment]
Alright, the issue is the combination of `object-fit: contain` and setting the body height to 100%. If you right-click your canvas and inspect it in dev tools, you'll see that it isn't the size shown in the browser. It's actually the entire page width / height. I've spent a while digging through the properties and Google but it appears as soon as a canvas becomes a replaced element then that's it, there's no way to read the scaled sizes the browser is using to display it, therefore I have no way of knowing where on the canvas they're clicking.

There are questions about this specific issue on S/O, none of which have answers! (https://stackoverflow.com/questions/40872814/how-to-get-scaled-size-of-html-canvas-when-using-object-fit-css-property) 

I even tried inspecting the MouseEvent to see if maybe that held some kind of offset, but get this - even if you click _outside_ of the visible canvas, you still get a mouse event from it! Clearly what the browser considers as being the canvas, and what it actually renders, are two different things.

In short, I cannot see how this can be fixed.

**Comment by samme** [comment]
I think the browser behavior is probably correct, but inconvenient for us.

**Comment by fariazz** [comment]
Oh wow thanks @photonstorm for looking into this! It is what it is then.

I feel we need some way to emulate the "SHOW_ALL" system from Phaser 2. It was a very handy mode.
